app/client/src/components/shared/Map.tsx

```diff
@@ -34,13 +36,16 @@ function Map({ layers = null, startingExtent = null }: Props) {
 
     // hide/show layers based on the visibleLayers object
     map?.layers.forEach((layer) => {
+      // ignore layers added from the add/save data widget
+      if (widgetLayers.some((l) => l.layer.id === layer.id)) return;
+
       if (visibleLayers.hasOwnProperty(layer.id)) {
         layer.visible = visibleLayers[layer.id as LayerId];
       } else {
         layer.visible = false;
       }
     });
```

Comment:
Good catch on this. The fix works great, but since `visibleLayers` has values for all layers we care about, we could also just remove the "else" clause:
```suggestion
    map?.layers.forEach((layer) => {
      if (visibleLayers.hasOwnProperty(layer.id)) {
        layer.visible = visibleLayers[layer.id as LayerId];
      }
    });
```

Reply:
I like it! Much simpler.